### ISSUE TITLE
Can't use subclass of Generic type in function

### ISSUE THREAD

**Issue by Dreamsorcerer** [python/mypy#11590]
```python
from typing import Generic, TypedDict, TypeVar

class A(TypedDict):
    foo: str

class B(A):
    bar: str

_T = TypeVar("_T", bound=A)

class Gen(Generic[_T]):
    def __init__(self, d: _T):
        self.d = d

def f(g: Gen[A]) -> None:
    g.d["foo"]

d = B(foo="yoo", bar="bar")
g = Gen(d)
f(g)  # error: Argument 1 to "f" has incompatible type "Gen[B]"; expected "Gen[A]"  [arg-type]
```

I think this should be safe and shouldn't cause a type error.

**Comment by erictraut** [comment]
I think mypy is correct to generate an error in this case. You have not marked `_T` as covariant, so the type must match exactly when used with `Gen`. If you add `covariant=True` to `_T`, the error goes away.

**Comment by Dreamsorcerer** [comment]
Ah, of course, I totally forgot about that. Thanks a lot!
